### PR TITLE
Use WallTimer to always publish pose and submaps.

### LINKED ISSUE #44: No submap_lists published if no data flows
The Cartographer node's global optimization can fall behind and if it does not see any more data coming in, it will stop updating the submap lists, though the optimization is still running. 

Maybe also related to #41

### PR DESCRIPTION
**Pull request by SirVer**
This makes the publishing independent from the inflow of sensor data and
is more in line with how the ROS world works.

Fixes #44.

### CODE REVIEW

File: cartographer_ros/src/cartographer_node_main.cc [hunk 1]
```diff
@@ -93,8 +93,8 @@ using carto::kalman_filter::PoseCovariance;
 // TODO(hrapp): Support multi trajectory mapping.
 constexpr int64 kTrajectoryId = 0;
 constexpr int kSubscriberQueueSize = 150;
-constexpr int kSubmapPublishPeriodInUts = 300 * 10000ll;  // 300 milliseconds
-constexpr int kPosePublishPeriodInUts = 5 * 10000ll;      // 5 milliseconds
+constexpr int kSubmapPublishPeriodMs = 300;
```
Comment (damonkohler) [inline cartographer_ros/src/cartographer_node_main.cc]: InMs

File: cartographer_ros/src/cartographer_node_main.cc [hunk 2]
```diff
@@ -553,6 +555,13 @@ void Node::Initialize() {
     occupancy_grid_thread_ =
         std::thread(&Node::SpinOccupancyGridThreadForever, this);
   }
+
+  wall_timers_.push_back(node_handle_.createWallTimer(
+      ros::WallDuration(kSubmapPublishPeriodMs / 1000.),
```
Comment (damonkohler) [inline cartographer_ros/src/cartographer_node_main.cc]: Maybe keep this constants around in seconds then?

File: cartographer_ros/src/cartographer_node_main.cc [hunk 3]
```diff
@@ -93,8 +93,8 @@ using carto::kalman_filter::PoseCovariance;
 // TODO(hrapp): Support multi trajectory mapping.
 constexpr int64 kTrajectoryId = 0;
 constexpr int kSubscriberQueueSize = 150;
-constexpr int kSubmapPublishPeriodInUts = 300 * 10000ll;  // 300 milliseconds
-constexpr int kPosePublishPeriodInUts = 5 * 10000ll;      // 5 milliseconds
+constexpr int kSubmapPublishPeriodMs = 300;
+constexpr int kPosePublishPeriodInMs = 5;
```
Comment (damonkohler) [inline cartographer_ros/src/cartographer_node_main.cc]: This should be configurable in the Lua.

**Comment by damonkohler** [comment]
LGTM

File: cartographer_ros/src/cartographer_node_main.cc (hunk 1 above)
Comment (SirVer) [inline cartographer_ros/src/cartographer_node_main.cc]: no longer applicable.

File: cartographer_ros/src/cartographer_node_main.cc (hunk 2 above)
Comment (SirVer) [inline cartographer_ros/src/cartographer_node_main.cc]: moved to lua.

File: cartographer_ros/src/cartographer_node_main.cc (hunk 3 above)
Comment (SirVer) [inline cartographer_ros/src/cartographer_node_main.cc]: done.

**Comment by damonkohler** [comment]
LGTM

**Comment by skohlbr** [comment]
This actually doesn't work correctly for all use cases. Currently, the transform is repeatedly published, but always with the last pose estimate timestamp. If pose estimates are internally generated at low rate (such as when aggregating spinning LIDAR data and matching only every few seconds or so), this means that the fixed rate publisher republishes the exact same tf data over and over, but as the timestamp does not progress, all system components waiting for tf connectivity stall until the SLAM pose estimate (and the timestamp alongside it) is updated. I'll open a PR with a fix for discussion.

**Pull request merged**